### DISCUSSION TITLE
[Feature] Extract Angular Schematics Support into a Dedicated Package Other Than `@nx/angular`

### DISCUSSION THREAD

**Discussion by Char2sGu** [nrwl/nx#22936] [Feature Requests]
Currently, Nx's support for Angular Schematics is kind of wierd. Nx itself has only partial support for Angular Schematics, which has led to confusing behaviors as described in #22625.

I was told that the Nx team is planning to completely extract the support for Angular Schematics from the Nx core into a new package. Angular Schematics will be supported only when the new package is present in `node_modules`, and Nx will use the new package to execute Angular Schematics.

Currently, the "new package" is `@nx/angular`, and I believe it is inappropriate:
1. If one decided to stick with Angular Schematics, they will likely not touch `@nx/angular` generators.
1. If one decided to go with `@nx/angular`, they will likely not need Angular Schematics support.
1. `@nx/angular` provides generators that use the same aliases as Angular Schematics, forcing the developer to specify the collection of a generator such as `nx g @schematics/angular:c` instead of `nx g c`, even if the developer only needs the generators from one collection (Angular Schematics or `@nx/angular` plugins).
1. `@nx/angular` is an Nx plugin, which should be responsible for only providing generators and executors. Support for Angular Schematics is not something a plugin should do.

Therefore, I propose that the support for Angular Schematics should be extracted into a package that is dedicated to provide the support. The package could be `@nx/schematics`.

When migrating from Angular CLI, Nx should install only the `@nx/schematics` package (please do not install `@nx/angular`), and continue to use Angular Schematics.

My personal experience of migrating from Angular CLI to Nx has been somewhat disappointing:

1. All my generator settings are targeting Angular Schematics. It is time consuming to migrate to `@nx/angular` generators.
1. `@nx/angular` generators are too opinionated and changes too much of my workspace configs and settings, destroying my own patterns for managing my workspace.
1. Even if I only want to use Angular Schematics, I have to install the `@nx/angular` plugin and specify the collect name `nx g @schematics/angular:c` instead of `nx g c`, as mentioned above.

If one wants to migrate from Angular CLI to Nx, the workspace can be scaled, and they would likely have their own pattern for managing the workspace. Nx should introduce minimal changes to allow the progressive transition to the Nx ecosystem.

Alternatively, explicitly ask whether to use `@nx/angular` or keep using schematics when migrating from Angular CLI.

**Comment by jaysoo** [comment]
Thanks for sharing your perspective. I'll discuss this with the team, as we want to provide a good experience for folks coming from Angular CLI.…
